credentials/credentials.go

```diff
@@ -334,3 +335,27 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 
 	return cfg.Clone()
 }
+
+// RequestInfo contains request data to be attached to the context passed along to GetRequestMetadata calls. This API is experimental.
```

Comment:
Let's also document on `GetRequestMetadata` in the `PerRPCCredentials` interface that the context will contain a `RequestInfo` retrievable via `RequestInfoFromContext`.

Reply:
Done. Let me know if you would like to change the wording.